projects/batfish-client/src/org/batfish/client/Client.java

```diff
@@ -315,6 +323,151 @@ private boolean answer(String questionTemplateName, String paramsLine,
+                  .getNodeType() + " doesn't match the type required for " +
+                  "variable type: " + variable.getType());
+         }
+      }
+      switch (variable.getType()) {
+         case BOOLEAN:
+            if (!value.isBoolean()) {
+               throw new BatfishException("Expected a boolean for " +
+                     "variable type: " + variable.getType());
+            }
+            break;
+         case COMPARATOR:
+            String textContent = value.textValue();
+            if (!(textContent.equals(">") || textContent.equals(">=") ||
+                  textContent.equals("==") || textContent.equals("!=") ||
+                  textContent.equals("<=") || textContent.equals("<"))) {
+               throw new BatfishException("Expected a comparator for " +
+                     "variable type: " + variable.getType());
+            }
+            break;
+         case INTEGER:
+            if (!value.isInt()) {
+               throw new BatfishException("Expected an Integer for " +
+                     "variable type: " + variable.getType());
+            }
+            break;
+         case IP:
+            //TODO: Need to double check isInetAddress()
+            if (!InetAddresses.isInetAddress(value.textValue())) {
+               throw new BatfishException("Not a valid ip address for " +
+                     "variable type: " + variable.getType());
+            }
+            break;
+         case IP_WILDCARD:
+            //TODO: need ip_WildCard validation check
+            break;
+         case JAVA_REGEX:
+            try {
+               Pattern.compile(value.textValue());
+            } catch (PatternSyntaxException e) {
+               throw new BatfishException("Expected a Java regular expression" +
+                     " for variable type: " + variable.getType());
+            }
+            break;
+         case JSON_PATH_REGEX:
+            validateJsonPathRegex(value.textValue());
+            break;
+         case PREFIX:
+            //TODO: need prefix validation check
+            break;
+         case PREFIX_RANGE:
+            //TODO: need prefix range validation check
+            break;
+         case STRING:
+            break;
+         case SUBRANGE:
+            //TODO: need prefix range validation check
+            break;
+         case JSON_PATH:
+            validateJsonPath(value);
+            break;
+      }
+   }
+
+   /**
+    * Validate that {@code jsonPathRegex}  contains a valid java regular
+    * expression of a json path(Starts with '/', ends with either '/' or
+    * '/i', contains a valid java regular expression between '/').
+    *  <p>If the content of {@code jsonPathRegex} is not a valid java regular
+    *  expression of a json path,
+    *  {@link BatfishException BatfishException} will be thrown.</p>
+    */
+   static void validateJsonPathRegex(String jsonPathRegex) throws
+         BatfishException {
```

Comment:
similar to the string construction comment above, it's best to break long lines semantically. That is, break them at places where the meaning or structure shifts, rather than in the middle of a construct. In this case:

```java
static void validateJsonPathRegex(String jsonPathRegex)
      throws BatfishException
```

would make more sense, because the `throws` is related to the `BatfishException`, not the function's argument list.